pages/index.js

```diff
@@ -52,6 +33,101 @@ export default function HomePage({
     end: null,
   });
   const [isCustomDatePickerOpen, setCustomDatePickerOpen] = useState(false);
+  const [isDeletingId, setIsDeletingId] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [editTransaction, setEditTransaction] = useState(null);
+  const [showForm, setShowForm] = useState(false);
+  const [activeSelectionId, setActiveSelectionId] = useState(null);
+
+  function handleOpenEditMode(transaction) {
+    setIsEditing(true);
+    setEditTransaction(transaction);
+    setIsModalOpen(true);
+  }
+
+  function openModal() {
+    setIsModalOpen(true);
+  }
+
+  function closeModal() {
+    setIsModalOpen(false);
+  }
+
+  function openSelection(selectionId) {
+    setActiveSelectionId(selectionId);
+  }
+
+  function closeSelection() {
+    setActiveSelectionId(null);
+  }
+
+  function handleOpenDeleteDialogue(id) {
+    setIsDeletingId(id);
+  }
+
+  function handleCancelDeleteDialogue() {
+    setIsDeletingId(false);
+  }
+
+  function toggleForm() {
+    setShowForm(!showForm);
+  }
+
+  async function handleAddTransaction(data) {
+    const response = await fetch("/api/transactions", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+
+    const newTransaction = await response.json();
+
+    mutateTransactions();
+    setSuccessMessage("Transaction successfully added!");
+    return newTransaction;
```

Comment:
I think you don't need to return the value here, since its the submit handler function which return values are typically ignored.